src/DGEList.py

```diff
@@ -40,29 +39,43 @@ class DGEList(object):
     _field_strip_re = re.compile(r'[\s"]+')
 
     def __init__(
-        self,
-        counts: Optional[np.matrix] = None,
-        samples: Optional[np.array] = None,
-        genes: Optional[np.array] = None,
-        norm_factors: Optional[np.array] = None,
-        group: Optional[np.array] = None,
-        to_remove_zeroes: Optional[bool] = True
+            self,
+            counts: Optional[np.ndarray] = None,
+            samples: Optional[np.array] = None,
+            genes: Optional[np.array] = None,
+            norm_factors: Optional[np.array] = None,
+            group: Optional[np.array] = None,
+            to_remove_zeroes: Optional[bool] = False,
+            filename: Optional[str]=None,
     ) -> None:
-        if counts is None:
-            counts = np.matrix(np.zeros(3))
-        if norm_factors is None:
-            norm_factors = np.ones(np.size(counts, 1))
 
         self.to_remove_zeroes = to_remove_zeroes
-        self.counts = counts
-        self.samples = samples
-        self.genes = genes
-        self.norm_factors = norm_factors
-        self.group = group
+        if filename:
+            if counts or samples or genes or norm_factors or group:
+                raise Exception("if filename is provided, you can't also provide other parameters")
+            self._counts = None
+            self.import_file(filename)
+
+        else:
+            if counts is None:
+                raise Exception("counts must be provided at init")
+
+            if norm_factors is None:
+                try:
+                    norm_factors = np.ones(np.size(counts, 1))
+                except IndexError:
+                    raise ValueError("counts must have more than one sample "
```

Comment:
Counts should be validated upon `__setitem__()` already. No need for try-except pattern.